.pre-commit-config.yaml

```diff
@@ -0,0 +1,61 @@
+repos:
```

Comment:
If we are going to have pre-commit hook, do we need link checks in our GitHub actions ?
https://github.com/kubeflow/katib/blob/master/.github/workflows/test-lint.yaml

Reply:
Nice catch! Should be resolved by: https://github.com/kubeflow/katib/pull/2242/commits/fa04b5b8ff48164b138d0ac1913dfcf61b6dd99e. We can probably move shellcheck to pre-commit eventually as well.